ext/ext.go

```diff
@@ -0,0 +1,160 @@
+package ext
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"sort"
+	"strings"
+	"sync"
+)
+
+// TODO: Make an ExtensionRegistry?
+//
+//nolint:gochecknoglobals
+var (
+	mx         sync.RWMutex
+	extensions = make(map[ExtensionType]map[string]*Extension)
+)
+
+// ExtensionType is the type of all supported k6 extensions.
+type ExtensionType uint8
+
+// All supported k6 extension types.
+const (
+	JSExtension ExtensionType = iota + 1
+	OutputExtension
+)
+
+func (e ExtensionType) String() string {
+	var s string
+	switch e {
+	case JSExtension:
+		s = "js"
+	case OutputExtension:
+		s = "out"
+	}
+	return s
+}
+
+// Extension is a generic container for any k6 extension.
+type Extension struct {
+	Name, Path, Version string
+	Type                ExtensionType
+	Mod                 interface{}
```

Comment:
Let's name it `Module` It's not longer than `Version` and doesn't harm but helps.

Reply:
Sure. Done in 6de4f4eb7.

I'm slightly against "module" (or "mod") though, since in the case of outputs, it's actually a function, but :man_shrugging: